### PR TITLE
Making size prop work for icon components using dash icon strings

### LINKED ISSUE #43724: Icon: Support `size` prop for dashicons
## What problem does this address?

The `size` prop does not work when the `icon` is a dashicon string.

## What is your proposed solution?

Add some handling (either to the Dashicon or the Icon component) so it adds a `font-size` style equivalent to the `size`.

### PR DESCRIPTION
**Pull request by matiasbenedetto**
## What?
Making size prop work for icon components using dash icon strings.

## Why?
Because currently is not working as the docs state: https://developer.wordpress.org/block-editor/reference-guides/components/icon/#size

## How?
Passing the prop to the child component and rendering the right styles/

## Testing Instructions

1. Import the icon component from `@wordpress/components` as:
`import { Icon }  from @wordpress/components;`
2. Render a dashicon like this:
`<Icon icon="plus-alt" size={10} />`
3. Check that the icon width and height are is10px.

### CODE REVIEW

File: packages/components/src/dashicon/index.js [hunk 1]
```diff
@@ -20,7 +22,17 @@ function Dashicon( { icon, className, ...extraProps } ) {
 		.filter( Boolean )
 		.join( ' ' );
 
-	return <span className={ iconClass } { ...extraProps } />;
+	// For retro-compatibility reasons (for example if people are overriding icon size with CSS), we add inline styles just if the size is different to the default
+	const style =
+		20 !== size
```
Comment (mirka) [inline packages/components/src/dashicon/index.js]: Should we also consider `"20"` (as string) to be the same as the default?

(No strong opinion on this one, it's just that a lot of people do things like `size="20"` and `size={ 20 }` kind of interchangeably in practice.)

File: packages/components/src/dashicon/index.js [hunk 2]
```diff
@@ -20,7 +22,17 @@ function Dashicon( { icon, className, ...extraProps } ) {
...
+			? {
+					fontSize: `${ size }px`,
+					width: `${ size }px`,
+					height: `${ size }px`,
+			  }
+			: {};
+
+	return <span className={ iconClass } style={ style } { ...extraProps } />;
```
Comment (mirka) [inline packages/components/src/dashicon/index.js]: I think we need to merge the incoming `style` prop so it works as before.

```jsx
// So something like this still works
<Dashicon icon="foo" style={{ background: 'yellow' }} />
```

File: packages/components/src/icon/test/index.js [hunk 3]
```diff
@@ -32,6 +32,16 @@ describe( 'Icon', () => {
 		);
 	} );
 
+	it( 'renders a dashicon with custom size', () => {
```
Comment (mirka) [inline packages/components/src/icon/test/index.js:35]: Thanks for the test ✨

**Review by mirka — COMMENTED** [review]
I like how this is thoughtful about back compat, thank you!

If you could also add a quick [changelog](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CHANGELOG.md), that would be great.

File: packages/components/src/dashicon/index.js (hunk 2 above)
Comment (matiasbenedetto) [reply packages/components/src/dashicon/index.js]: Goog catch, thanks! Done.

File: packages/components/src/dashicon/index.js (hunk 1 above)
Comment (matiasbenedetto) [reply packages/components/src/dashicon/index.js]: I used `!=` to catch both 20 and "20". I added an inline comment to explain that.

**Comment by matiasbenedetto** [comment]
Thanks for the detailed review @mirka !
I think I addressed all your comments in the latest commits.

**Review by mirka — APPROVED** [review]
Perfect, thank you!

**Pull request merged**